### DISCUSSION TITLE
Newbie questions

### DISCUSSION THREAD

**Discussion by tlc** [jertel/elastalert2#678] [Q&A] [answered]
I'm just getting started on alerting for ElasticSearch.

* Does elastalert2 have any contenders for best, most viable, open-source alerting for ElasticSearch?
* The yelp/elastalert README links to a _very old_ kibana plugin, but elastalert2 makes no mention of that.
  * Are there any recommended replacements? 
  * Besides other alerting paths, I'd like to make alerts show up (pop up)  _on_ kibana pages.  Is a plugin needed for that?
  * Also, does elastalert2 have a web GUI for creating new alerts?  Or is/was that function of the plugin?

Thanks for keeping this alive!

**Comment by jertel** [comment, marked as answer]
> * Does elastalert2 have any contenders for best, most viable, open-source alerting for ElasticSearch?

Both OpenSearch and Elasticsearch now include their own alerting systems. There are pros/cons of both, and with the Elasticsearch alerting you must purchase an Elastic license for many of the alerting features.

> * The yelp/elastalert README links to a _very old_ kibana plugin, but elastalert2 makes no mention of that.

As it's a separate project and appears to be abandoned, I chose not to reference it from this project. I've never used it and have no further information on it.

>   * Are there any recommended replacements?

Replacement for what? The Kibana Plugin? Not that I know of.

>   * Besides other alerting paths, I'd like to make alerts show up (pop up)  _on_ kibana pages.  Is a plugin needed for that?

I don't know what you mean here. Perhaps a screen mockup would help? But my guess is there is no support for this.

>   * Also, does elastalert2 have a web GUI for creating new alerts?  Or is/was that function of the plugin?

Check out this project: https://github.com/johnsusek/praeco

**Reply by nsano-rururu** [reply]
>The yelp/elastalert README links to a very old kibana plugin, but elastalert2 makes no mention of that.

look
https://github.com/Karql/elastalert2-server
https://github.com/Karql/elastalert-kibana-plugin

**Reply by nsano-rururu** [reply]
>Check out this project: https://github.com/johnsusek/praeco

It does not support direct modification of rule files, so we recommend that you do not use it if you want to modify it directly.
